### DISCUSSION TITLE
A media card

### DISCUSSION THREAD

**Discussion by pdhruv93** [shannonhochkins/ha-component-kit#39] [Ideas]
A card to control media players like Chromecast or nest devices

**Comment by shannonhochkins** [comment]
Definitely on my list, but I've never used media players to play specific media, would appreciate a bit of insight if you have some! 

If not, I can always look at existing media cards for home assistant and go from there

**Reply by pdhruv93** [reply]
This is the entity details I see in the developer tools:
![Screenshot 2023-09-06 at 11 33 26](https://github.com/shannonhochkins/ha-component-kit/assets/38987871/043fb1f5-52c2-46fa-9620-045101f6905f)

we can have just the UI component with controls like
![Screenshot 2023-09-06 at 11 29 45](https://github.com/shannonhochkins/ha-component-kit/assets/38987871/ff8e7da0-3579-48c5-9b55-1b0df57da183)

Please feel free to ping me if you need any help(if you have any discord or slack), I would be happy to help you are doing great work.

**Reply by shannonhochkins** [reply]
Thanks man - i've seen that card before, i'm just more curious about how you "pick" media to play! is it stored on home assistant under the media section or something?

**Reply by pdhruv93** [reply]
I usually play using google assistant or cast from my mobile to the nest devices and those media are reflected on the HA. 
HA also has the pick media but that's not very good/evolved(only allows you to play certain radio stations and not your own file from system) and I never use it. So I think you can skip the pick media part.

**Reply by shannonhochkins** [reply]
Just had a quick look, seems promising, was able to render the curren picture, skip, set the volume pretty easily! Shouldn't be too hard to create something like the above :)

**Reply by shannonhochkins** [reply]
After a bit more digging, it appears (at least over sockets) or maybe it's just YouTube Music, the duration doesn't update, so I won't be able to have a "progress" bar of the current song, might have to be a bit more of a basic component but it's mostly doable

**Reply by shannonhochkins** [reply]
Nevermind, find a post about it, they do it intentially so they're not writing to the entitiy continuously, but they also provide a media_position_updated_at which allows you to calculate a sliding timer when the device is "playing" got it workin!

**Reply by shannonhochkins** [reply]
![Screenshot 2023-09-06 at 9 13 21 pm](https://github.com/shannonhochkins/ha-component-kit/assets/5105002/85c8f706-4bbb-4aa0-891a-68d23c2203fa)

Managed to get everything working, need to cleanup the UI a lot obviously, but seems to be working nicely! 

Going to require a lot of testing, so might be a week or so before i can get this one out in my free time between work mate

**Reply by shannonhochkins** [reply]
@pdhruv93 

Hi mate - just wondering, do you use the "join" method so you can play to multiple speakers at once?

I only have the nest minis and nest hubs at home and it seems they don't support the joining method unlike sonos speakers

Would be good to understand if you do as i may need you to test this feature for me :D

**Reply by pdhruv93** [reply]
Hi. Unfortuantely the "Speaker Group" feature does not works on Google devices and I am more of a Google person and don't have other devices, so cannot check it.

**Reply by shannonhochkins** [reply]
Bugger, that's okay - might just have to put a disclaimer with it that the join method may not work, here's what i have so far:

The track title is a marquee, so it animates if the title is too long - still have a fair bit more to go unfortunately
![Screenshot 2023-09-08 at 5 35 02 pm](https://github.com/shannonhochkins/ha-component-kit/assets/5105002/88d17bb3-4b6a-4ff8-abbf-e4dea5dcf137)

**Reply by pdhruv93** [reply]
Looks nice. Let me know when it is available for use. Will put it on my dashboard.

**Comment by shannonhochkins** [comment]
@pdhruv93 I've just released a [massive change](https://github.com/shannonhochkins/ha-component-kit/pull/45) to support this new card, if you wouldn't mind, can you install the latest core & components and give me some feedback?

All documentation is live and up on the storybook documentation :)

**Reply by pdhruv93** [reply]
I will have a look and let you know by tomorrow evening.

**Reply by pdhruv93** [reply]
@shannonhochkins works nicely. All functions working!

**Reply by shannonhochkins** [reply]
Hell yeah, you happy with it?

**Reply by pdhruv93** [reply]
Yes, looks very clean and minimalistic. Atleast now I have all possible cards on my dashboard!

**Reply by shannonhochkins** [reply]
Awesome, I'll close this discussion now :)
